test/hotspot/gtest/nmt/test_nmt_malloclimit.cpp

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2022 SAP SE. All rights reserved.
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+#include "precompiled.hpp"
+#include "memory/allocation.hpp"
+#include "runtime/os.hpp"
+#include "services/mallocLimit.hpp"
+#include "services/memTracker.hpp"
+#include "services/nmtCommon.hpp"
+#include "utilities/debug.hpp"
+#include "utilities/globalDefinitions.hpp"
+#include "utilities/ostream.hpp"
+
+#include "testutils.hpp"
+#include "unittest.hpp"
+
+// Tests here just test the MallocLimit option parser. They are complemented
+// by more extensive jtreg tests (runtime/NMT/TestMallocLimit.java)
+static bool compare_limits(const malloclimit_t* a, const malloclimit_t* b) {
+  return a->v == b->v && a->f == b->f;
+}
+
+static bool compare_sets(const MallocLimitSet* a, const MallocLimitSet* b) {
+  if (compare_limits(a->global_limit(), b->global_limit())) {
+    for (int i = 0; i < mt_number_of_types; i++) {
+      if (!compare_limits(a->category_limit(NMTUtil::index_to_flag(i)),
+                          b->category_limit(NMTUtil::index_to_flag(i)))) {
+        return false;
+      }
+    }
+  }
+  return true;
+}
+
+static void test(const char* s, const MallocLimitSet& expected) {
+//  LOG_HERE("%s", s);
```

Comment:
Seems like a left over from development.

Reply:
I often leave these in at neuralgic points to quickly analyze problems, they use the LOG_HERE macro in testutils.hpp. But I can remove it.